src/player_input_manager.js

```diff
@@ -45,9 +45,18 @@ Keydrown.RIGHT.up(() => { keyboardButtons.right = false })
 
 export const playerInputUpdate = () => {
     Keydrown.tick()
-
-    let stickX = 0, stickY = 0
-
+    let stickX = 0, stickY = 0, gamepad, contA, contB, contStart, contZ
+    if (navigator.getGamepads) {
+        gamepad = navigator.getGamepads()[0];
+    }
+    if (gamepad) {
+        stickX = gamepad.axes[0];
+        stickY = gamepad.axes[1] * -1;
+        contA = gamepad.buttons[0].touched
+        contB = gamepad.buttons[2].touched
+        contStart = gamepad.buttons[9].touched
```

Comment:
these button offsets seem to work well for the xbox controllers i tested on, but i guess it won't work for everything